chain/ethereum/gas.go

```diff
@@ -3,31 +3,165 @@ package ethereum
 import (
 	"context"
 	"fmt"
+	"math/big"
+	"sort"
+
 	"github.com/renproject/pack"
 )
 
+const (
+	// FeeHistoryBlocks specifies how many blocks to consider for priority fee estimation
+	FeeHistoryBlocks = 10
+	// FeeHistoryPercentile specifies the percentile of effective priority fees to include
+	FeeHistoryPercentile = 5
+	// FallbackMaxFeePerGas is the fallback value used when MaxFeePerGas cannot be calculated
+	FallbackMaxFeePerGas = 20000000000
+)
+
+var (
+	// PriorityFeeEstimationTrigger specifies which base fee to trigger priority fee estimation at
+	PriorityFeeEstimationTrigger = big.NewInt(100000000000) // WEI
+	// DefaultPriorityFee is returned if above trigger is not met
+	DefaultPriorityFee          = big.NewInt(3000000000)
+	PriorityFeeIncreaseBoundary = big.NewInt(200) // %
+)
+
+type feeHistoryResult struct {
+	OldestBlock int        `json:"oldestBlock"`
+	Reward      [][]string `json:"reward"`
+}
+
+type GasOptions struct {
+	FeeHistoryBlocks             uint64
+	FeeHistoryPercentile         uint64
+	FallbackMaxFeePerGas         uint64
+	PriorityFeeEstimationTrigger *big.Int
+	DefaultPriorityFee           *big.Int
+	PriorityFeeIncreaseBoundary  *big.Int
+}
+
 // A GasEstimator returns the gas price and the provide gas limit that is needed in
 // order to confirm transactions with an estimated maximum delay of one block.
 type GasEstimator struct {
-	client *Client
+	client  *Client
+	options *GasOptions
 }
 
 // NewGasEstimator returns a simple gas estimator that fetches the ideal gas
-// price for a ethereum transaction to be included in a block
-// with minimal delay.
-func NewGasEstimator(client *Client) *GasEstimator {
+// price for an ethereum transaction to be included in a block
+// with minimal delay. If opts is nil it uses the default gas settings
+func NewGasEstimator(client *Client, opts *GasOptions) *GasEstimator {
+	if opts != nil {
+		return &GasEstimator{
+			client:  client,
+			options: opts,
+		}
+	}
```

Comment:
Instead of allowing `nil` as a `GasOptions` parameter, you could declare another function:
```go
func DefaultGasOptions() GasOptions {
	return GasOptions{
		FeeHistoryBlocks,
		FeeHistoryPercentile,
		FallbackMaxFeePerGas,
		PriorityFeeEstimationTrigger,
		DefaultPriorityFee,
		PriorityFeeIncreaseBoundary,
	}
}
```